--- newlib-1.13.0.orig/config.sub	2004-11-15 17:18:39.000000000 -0800
+++ newlib-psp/config.sub	2005-06-03 13:48:01.000000000 -0700
@@ -253,6 +253,7 @@
 	| mipsisa64sb1 | mipsisa64sb1el \
 	| mipsisa64sr71k | mipsisa64sr71kel \
 	| mipstx39 | mipstx39el \
+	| mipsallegrex | mipsallegrexel \
 	| mn10200 | mn10300 \
 	| msp430 \
 	| ns16k | ns32k \
@@ -326,6 +327,7 @@
 	| mipsisa64sb1-* | mipsisa64sb1el-* \
 	| mipsisa64sr71k-* | mipsisa64sr71kel-* \
 	| mipstx39-* | mipstx39el-* \
+	| mipsallegrex-* | mipsallegrexel-* \
 	| mmix-* \
 	| msp430-* \
 	| none-* | np1-* | ns16k-* | ns32k-* \
@@ -665,6 +667,10 @@
 		basic_machine=m68k-atari
 		os=-mint
 		;;
+	psp)
+		basic_machine=mipsallegrexel-unknown
+		os=-elf
+		;;
 	mips3*-*)
 		basic_machine=`echo $basic_machine | sed -e 's/mips3/mips64/'`
 		;;
--- newlib-1.13.0.orig/configure	2004-12-16 11:51:28.000000000 -0800
+++ newlib-psp/configure	2005-06-14 21:58:20.783668856 -0700
@@ -1507,6 +1507,9 @@
   mips*-*-linux*)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss"
     ;;
+  mipsallegrex*-*-*)
+    noconfigdirs="$noconfigdirs target-libgloss"
+    ;;
   mips*-*-*)
     noconfigdirs="$noconfigdirs gprof ${libgcj}"
     ;;
--- newlib-1.13.0.orig/configure.in	2004-12-16 11:51:28.000000000 -0800
+++ newlib-psp/configure.in	2005-06-14 21:57:15.303623336 -0700
@@ -716,6 +716,9 @@
   mips*-*-linux*)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss"
     ;;
+  mipsallegrex*-*-*)
+    noconfigdirs="$noconfigdirs target-libgloss"
+    ;;
   mips*-*-*)
     noconfigdirs="$noconfigdirs gprof ${libgcj}"
     ;;
--- newlib-1.13.0.orig/newlib/libc/ctype/ctype_.c	2003-05-13 02:46:47.000000000 -0700
+++ newlib-psp/newlib/libc/ctype/ctype_.c	2005-05-23 23:40:50.000000000 -0700
@@ -88,7 +88,7 @@
 extern _CONST char __declspec(dllexport) _ctype_[1 + 256] __attribute__ ((alias ("_ctype_b+127")));
 _CONST char __declspec(dllexport) *__ctype_ptr = _ctype_b + 128;
 #else
-extern _CONST char _ctype_[1 + 256] __attribute__ ((alias ("_ctype_b+127")));
+extern _CONST char _ctype_[1 + 256]; // __attribute__ ((alias ("_ctype_b+127")));
 _CONST char *__ctype_ptr = _ctype_b + 128;
 #endif
 
