--- newlib-1.13.0.orig/config.sub	2004-11-15 17:18:39.000000000 -0800
+++ newlib-psp/config.sub	2005-06-24 17:04:02.000000000 -0700
@@ -253,6 +253,7 @@
 	| mipsisa64sb1 | mipsisa64sb1el \
 	| mipsisa64sr71k | mipsisa64sr71kel \
 	| mipstx39 | mipstx39el \
+	| mipsallegrex | mipsallegrexel \
 	| mn10200 | mn10300 \
 	| msp430 \
 	| ns16k | ns32k \
@@ -326,6 +327,7 @@
 	| mipsisa64sb1-* | mipsisa64sb1el-* \
 	| mipsisa64sr71k-* | mipsisa64sr71kel-* \
 	| mipstx39-* | mipstx39el-* \
+	| mipsallegrex-* | mipsallegrexel-* \
 	| mmix-* \
 	| msp430-* \
 	| none-* | np1-* | ns16k-* | ns32k-* \
@@ -665,6 +667,10 @@
 		basic_machine=m68k-atari
 		os=-mint
 		;;
+	psp)
+		basic_machine=mipsallegrexel-psp
+		os=-elf
+		;;
 	mips3*-*)
 		basic_machine=`echo $basic_machine | sed -e 's/mips3/mips64/'`
 		;;
--- newlib-1.13.0.orig/configure	2004-12-16 11:51:28.000000000 -0800
+++ newlib-psp/configure	2005-06-19 18:09:12.000000000 -0700
@@ -1507,6 +1507,9 @@
   mips*-*-linux*)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss"
     ;;
+  mipsallegrex*-*-*)
+    noconfigdirs="$noconfigdirs target-libgloss"
+    ;;
   mips*-*-*)
     noconfigdirs="$noconfigdirs gprof ${libgcj}"
     ;;
--- newlib-1.13.0.orig/configure.in	2004-12-16 11:51:28.000000000 -0800
+++ newlib-psp/configure.in	2005-06-19 18:09:12.000000000 -0700
@@ -716,6 +716,9 @@
   mips*-*-linux*)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss"
     ;;
+  mipsallegrex*-*-*)
+    noconfigdirs="$noconfigdirs target-libgloss"
+    ;;
   mips*-*-*)
     noconfigdirs="$noconfigdirs gprof ${libgcj}"
     ;;
--- newlib-1.13.0.orig/newlib/configure.host	2004-10-05 12:44:24.000000000 -0700
+++ newlib-psp/newlib/configure.host	2005-07-06 14:21:16.030259592 -0700
@@ -571,6 +571,11 @@
   mips*-*-elf*)
 	default_newlib_io_long_long="yes"
 	newlib_cflags="${newlib_cflags} -DMISSING_SYSCALL_NAMES"
+	case "${host}" in
+	  mipsallegrex*-psp-*)
+		syscall_dir=syscalls
+		newlib_cflags="${newlib_cflags} -G0 -DPREFER_SIZE_OVER_SPEED -DCOMPACT_CTYPE -DCLOCK_PROVIDED -UMISSING_SYSCALL_NAMES" ;;
+	esac
 	;;
   mmix-*)
 	syscall_dir=syscalls
